bounty_hunter/src/models.rs

```diff
@@ -1,9 +1,277 @@
-use super::schema::status;
-
-#[derive(Queryable, Serialize, Deserialize, Debug, Insertable)]
-#[table_name = "status"]
-pub struct Status {
-    pub ip: String,
-    pub mac: String,
-    pub balance: String,
+use clarity::Signature;
+use failure::Error;
+use num256::Uint256;
+use num_traits::Zero;
+
+use std::convert::From;
+
+use schema::states;
+
+#[derive(Deserialize, Serialize, Debug, Clone, PartialEq, Eq)]
+pub struct ChannelState {
+    pub channel_id: Uint256,
+    pub nonce: Uint256,
+
+    pub balance_a: Uint256,
+    pub balance_b: Uint256,
+
+    pub signature_a: Option<Signature>,
+    pub signature_b: Option<Signature>,
+}
+
+/// A helper type that prepares a [ChannelState](ChannelState) for storage by turning the contents into types
+/// easy to grasp for `diesel`.
+#[derive(Queryable, Debug, Clone, PartialEq, Eq, Default)]
+pub struct ChannelStateRecord {
+    pub id: i32,
```

Comment:
We want to use u32 for ids (or even u64) just to be safe

Reply:
Great point, although it seems that the SQLite3 backend can only do 32-bit signed. Will research though.

Reply:
Managed to bump to 64-bit, still signed though. No idea if `diesel`'s smart enough to deal with that.